### PR TITLE
LL-6359  - SWAP - Refactor and Optimizations

### PR DESCRIPTION
**Pull request by jelbaz-ledger**
## 🦒 Context (issues, jira)

[LL-6359]

- Refactored how we handle the state throughout the swap (removed duplicates and useless state).
- Eliminated `useEffect` chains causing extra renders.
- Memoized components and computations / callbacks to prevent excessive rendering.
- Fixed target account drawer displaying only parent accounts.

## 💻  Description / Demo (image or video)

**Before**

https://user-images.githubusercontent.com/86958797/133817168-1ae39abb-e895-4935-aec2-1c87af006576.mp4

**After**

https://user-images.githubusercontent.com/86958797/133817179-246fb581-ae39-498e-bcc7-2f85b8624852.mp4

## 🖤  Expectations to reach

PR must pass CI, merge develop if conflicts, do not force push. Thanks!

- **on QA**: at least one of these two checkboxes must be checked:
  - [ ] a specific test planned is defined on Jira
  - [ ] this PR is covered by automatic UI test
- **on delivery**: at least one of these two checkboxes must be checked: 
  - [x] Option 1: **no impact**: The changes of this PR have ZERO impact on the userland (invisible for users)
  - [ ] Option 2: **atomic delivery**: the changes is atomic and complete (no partial delivery)

PR must pass CI, merge develop if conflicts, do not force push. Thanks!

[LL-6359]: https://ledgerhq.atlassian.net/browse/LL-6359

### CODE REVIEW

**Review by jgiamporcaro-ledger — APPROVED** [review]
LGTM 🔥

File: src/renderer/screens/exchange/Swap2/Form/FormSummary/SectionRate.js [hunk 1]
```diff
@@ -82,4 +90,4 @@ const SectionRate = ({ swapTransaction, provider }: Props) => {
   );
 };
 
-export default SectionRate;
```
Comment (juan-cortes) [inline src/renderer/screens/exchange/Swap2/Form/FormSummary/SectionRate.js:85]: I feel wrapping this component in `React.memo` does more harm than good considering every time we change anything, the props change 🤔

File: src/renderer/screens/exchange/Swap2/utils/shared/index.js [hunk 2]
```diff
@@ -1,9 +1,30 @@
...
 
 export type KYCStatus = $Keys<typeof KYC_STATUS>;
+
+export const pickExchangeRate = (
+  exchangeRates: ExchangeRate[],
+  exchangeRate: ExchangeRate,
+  setExchangeRate: (?ExchangeRate) => void,
+) => {
+  const hasRates = exchangeRates && exchangeRates.length > 0;
+  // If a the user picked an exchange rate before, try to select the new one that matches.
```
Comment (juan-cortes) [inline src/renderer/screens/exchange/Swap2/utils/shared/index.js:19]:
```suggestion
  // If the user picked an exchange rate before, try to select the new one that matches.
```

File: src/renderer/screens/exchange/Swap2/utils/shared/index.js [hunk 3]
```diff
@@ -1,9 +1,30 @@
...
 };
 
 export type KYCStatus = $Keys<typeof KYC_STATUS>;
+
+export const pickExchangeRate = (
+  exchangeRates: ExchangeRate[],
+  exchangeRate: ExchangeRate,
+  setExchangeRate: (?ExchangeRate) => void,
+) => {
+  const hasRates = exchangeRates && exchangeRates.length > 0;
```
Comment (juan-cortes) [inline src/renderer/screens/exchange/Swap2/utils/shared/index.js:18]:
```suggestion
  const hasRates = exchangeRate?.length > 0;
```

**Review by juan-cortes — APPROVED** [review]

**Pull request merged**